airflow/jobs/base_job_runner.py

```diff
@@ -17,28 +17,31 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Generic, TypeVar
 
 from airflow.utils.session import NEW_SESSION, provide_session
 
 if TYPE_CHECKING:
     from sqlalchemy.orm import Session
 
     from airflow.jobs.job import Job
+    from airflow.serialization.pydantic.job import JobPydantic
 
+JobOrJobPydantic = TypeVar("JobOrJobPydantic", Job, JobPydantic)
```

Comment:
I think this needs to be in the `TYPE_CHECKING` block to avoid NameError at runtime

Reply:
yep